internal/controllers/topology/cluster/structuredmerge/serversidepathhelper.go

```diff
@@ -50,6 +50,8 @@ func NewServerSidePatchHelper(original, modified client.Object, c client.Client,
 		{"kind"},
 		{"metadata", "name"},
 		{"metadata", "namespace"},
+		// uid is optional for a server side apply intent but sets the expectation of an object getting created or a specific one updated.
+		{"metadata", "uid"},
```

Comment:
Do we still need this?

Reply:
Yes, it is required, otherwise it gets filters out and we would always set it to empty.

I preferred to add it here to only work on the filtered unstructured objects to have an easier nil check.

Reply:
How does it get filtered out given that we use the helperOptions in filterObject only before we set the uid?